### PR TITLE
SISRP-22857 - 6.0 - Class Enrollment - Refactor Back-end and Front-end

### PR DESCRIPTION
**Pull request by redconfetti**
https://jira.berkeley.edu/browse/SISRP-22857

### CODE REVIEW

File: src/assets/templates/widgets/enrollment/plan.html [hunk 1]
```diff
@@ -1,67 +1,65 @@
...
 
-      <div data-ng-if="enrollment.academicPlan.updateLink && enrollmentTerm.academicPlan.classes.length">
-        <div class="cc-flex cc-flex-space-between">
-          <div>
-            <strong>Current Academic Plan for <span data-ng-bind="enrollmentTerm.termDescr"></span></strong>
-          </div>
+  <div data-ng-if="api.user.profile.actAsOptions.canSeeCSLinks">
+    <div data-ng-if="enrollment.academicPlan.updateLink && !enrollmentInstruction.academicPlan.classes.length">
+      Build your multi-year
+      <a data-cc-campus-solutions-link-directive="enrollment.academicPlan.updateLink.url"
```
Comment (redconfetti) [inline src/assets/templates/widgets/enrollment/plan.html]: I can see that I missed updating these references to the academic plan data.

File: app/models/my_academics/class_enrollments.rb [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+        type_key = [type_code, career_code]
+        types[type_key] = { instruction_type_code: type_code, career_code: career_code, academic_plans: [] } if types[type_key].blank?
+        types[type_key][:academic_plans] << plan
+      end
+      types
+    end
+
+    def get_enrollment_instruction_type_code(plan)
+      type_codes = []
+      ENROLLMENT_INSTRUCTION_TYPES.keys.each do |cpp_category|
```
Comment (redconfetti) [inline app/models/my_academics/class_enrollments.rb]: Ray says : If we use `each`, we should get key/value pairs, which can simplify this code.

**Comment by raydavis** [comment]
As said in the walkthrough, @redconfetti might reconsider some naming and some minor code simplification, but basically this looks great. Thank you!

File: app/models/my_academics/class_enrollments.rb (hunk 2 above)
Comment (redconfetti) [inline app/models/my_academics/class_enrollments.rb]: @raydavis - I fixed this in #5657

**Pull request merged**